datafusion/src/optimizer/timestamp_evaluation.rs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Optimizer rule to replace timestamp expressions to constants.
+//! This saves time in planning and executing the query.
+use crate::error::Result;
+use crate::logical_plan::{Expr, LogicalPlan};
+use crate::optimizer::optimizer::OptimizerRule;
+
+use super::utils;
+use crate::execution::context::ExecutionProps;
+use crate::physical_plan::functions::BuiltinScalarFunction;
+use crate::scalar::ScalarValue;
+use chrono::{DateTime, Utc};
+
+/// Optimization rule that replaces timestamp expressions with their values evaluated
```

Comment:
What would you think about extending the `constant_folding` optimizer rule instead of adding an entirely new rule?

I think you could probably add a case to the expression rewriter in https://github.com/apache/arrow-datafusion/blob/master/datafusion/src/optimizer/constant_folding.rs#L122 for `Expr::ScalarFunction { fun: BuiltinScalarFunction::Now}}`  and avoid the need for all the code in this pass for traversing the expression tree.